thunder/tests/test_jit_general.py

```diff
@@ -991,3 +991,59 @@ def forward(self, x):
     # We can't directly compare state_dict - https://github.com/Lightning-AI/lightning-thunder/issues/647
     torch.testing.assert_close(thunder_module._overrides_parameters["param"], m.param)
     torch.testing.assert_close(thunder_module._overrides_buffers["buffer"], m.buffer)
+
+
+@pytest.mark.parametrize("prefix", ("", "foo"), ids=("prefix=", "prefix=foo"))
+@pytest.mark.parametrize("recurse", (True, False), ids=("recurse=True", "recurse=False"))
+@pytest.mark.parametrize(
+    "remove_duplicate",
+    (
+        False,
+        pytest.param(
+            True,
+            marks=pytest.mark.xfail(
+                raises=(AssertionError),
+                reason="We can't differentiate between shared buffers and parameters",
```

Comment:
Maybe there is a way for this?

Reply:
To my mind, we could record the Ids of tensors we proxy, then we could know?

Reply:
Yes, but we could get the deduplicated list inside the lookaside and then filter by that. :)